DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,312 @@
+```
+
+## Before Opening a Pull Request
+
+Make sure you have an issue on GitHub. All PRs require an issue. Describe the problem you're trying to solve and how you plan to address it. If the issue has the `require/hep` label, a [HEP (Harvester Enhancement Proposal)](./enhancements) is required for discussion before implementation.
+
+The general workflow is:
+- Create or find an issue.
+- Discuss why we need to fix it, what the goal is, and how to solve it.
+- Open a pull request for an HEP if needed.
+- Open a pull request to solve the issue.
+- Test the solution.
+
+### How to Find an Issue to Work On
+
+All issues are tracked in the harvester/harvester repository. Issues with a milestone or a "good first issue" label are ready for development. You can use [this filter](https://github.com/harvester/harvester/issues?q=is%3Aissue%20state%3Aopen%20has%3Amilestone) and [this filter](https://github.com/harvester/harvester/issues?q=state%3Aopen%20label%3A%22good%20first%20issue%22) to find open issues with a milestone.
+
+If an issue is already assigned but you want to work on it, leave a comment asking to be assigned or to coordinate with the current assignee. Discussion is welcome.
+
+### Branch Strategy
+
+The branch strategy for the harvester/harvester repository is:
+
+- Create a pull request that targets the `master` or `main` branch. Some older repositories still use the `master` branch.
+
+For more detailed information, see the [Branch Strategy Wiki](https://github.com/harvester/harvester/wiki/Branch-Strategy).
+
+### Code Style
+
+The code must be linted with `golangci-lint`. You can manually run the linter, configure your IDE with the [config](./.golangci.yaml), or run the following command to validate:
+
+```bash
+make validate
+```
+
+### Commit Message Format
+
+We don't enforce a strict commit message format; any reasonable format is acceptable. One recommendation is [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
+
+## After Opening a Pull Request
+
+- Provide a test plan in the PR description. Describe how to test and the expected results. A short demo recording is welcome.
+- Ensure all checks pass in the PR.
+
+## After All PRs Are Merged
+
+We use the GitHub Project to manage our issues (see [Issue Management](https://github.com/harvester/harvester/wiki/Issue-Management)). Once all PRs are merged, we'll move the issue status to "Ready for Test". The bot will then create a comment titled "Pre Ready-For-Testing Checklist". Please fill out all necessary information in that comment.
+
+Then backport your changes to supported stable branches via backport pull requests if needed. Please see [Backport Policy](https://github.com/harvester/harvester/wiki/Backport-Policy) for more information. We usually backport PRs to the `v1.x` branch. [This PR](https://github.com/harvester/harvester/pull/8810) is a backport example.
```

Comment:
I think we should either simplify this or leave it out for now. I don't expect an community contributor to know what needs to be backported, which labels are involved etc. Too much friction for a good dev UX. Ultimately, the maintainers should be the ones deciding and managing backports.

Maybe we can say something like this:

"Depending on the severity and scope of your changes, discuss with the maintainers to decide if your pull requests need to be backport to older supported versions of Harvester."

Reply:
I agree with that. Even though a new joiner of Harvester doesn't know the rules. It's better to discuss the owner or maintainers. I'll put your words on it.